resources/eventing/charts/controller/templates/pre-upgrade-job.yaml

```diff
@@ -0,0 +1,91 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels: {{- include "eventing.preUpgrade.labels" . | nindent 4 }}
+  name: {{ template "eventing.fullname" . }}-pre-upgrade
+  annotations:
+    helm.sh/hook-weight: "0"
+    helm.sh/hook: "pre-upgrade"
+    helm.sh/hook-delete-policy: "before-hook-creation,hook-succeeded"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels: {{- include "eventing.preUpgrade.labels" . | nindent 4 }}
+  name: {{ template "eventing.fullname" . }}-pre-upgrade
+  annotations:
+    helm.sh/hook-weight: "0"
+    helm.sh/hook: "pre-upgrade"
+    helm.sh/hook-delete-policy: "before-hook-creation,hook-succeeded"
+rules:
+  - apiGroups: ["apps"]
+    resources: ["deployments", "deployments/scale"]
+    resourceNames: ["eventing-controller", "eventing-publisher-proxy"]
+    verbs: ["get", "list", "patch", "delete"]
+  - apiGroups: [""]
+    resources: ["secrets", "configmaps"]
+    verbs: ["get", "list"]
+  - apiGroups: ["eventing.kyma-project.io"]
+    resources: ["subscriptions", "eventingbackends"]
+    verbs: ["get", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels: {{- include "eventing.preUpgrade.labels" . | nindent 4 }}
+  name: {{ template "eventing.fullname" . }}-pre-upgrade
+  annotations:
+    helm.sh/hook-weight: "1"
+    helm.sh/hook: "pre-upgrade"
+    helm.sh/hook-delete-policy: "before-hook-creation,hook-succeeded"
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ template "eventing.fullname" . }}-pre-upgrade
+subjects:
+  - kind: ServiceAccount
+    name: {{ template "eventing.fullname" . }}-pre-upgrade
+    namespace: {{ .Release.Namespace }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  labels: {{- include "eventing.preUpgrade.labels" . | nindent 4 }}
+  name: {{ template "eventing.fullname" . }}-pre-upgrade
+  annotations:
+    helm.sh/hook-weight: "2"
+    helm.sh/hook: "pre-upgrade"
+    helm.sh/hook-delete-policy: "before-hook-creation,hook-succeeded"
+spec:
+  backoffLimit: 5
+  template:
+    metadata:
+      name: {{ template "eventing.fullname" . }}-pre-upgrade
+      labels: {{- include "eventing.preUpgrade.labels" . | nindent 8 }}
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      serviceAccountName: {{ template "eventing.fullname" . }}-pre-upgrade
+      restartPolicy: Never
+      containers:
+        - name: {{ template "eventing.fullname" . }}-pre-upgrade
+          image: "{{ .Values.global.image.repository }}/{{ .Values.preUpgradeJob.image.name }}:{{ .Values.preUpgradeJob.image.tag }}"
+          imagePullPolicy: "{{ .Values.preUpgradeJob.image.pullPolicy }}"
+          terminationMessagePolicy: "FallbackToLogsOnError"
+          {{- if .Values.global.securityContext }}
+          securityContext: {{- toYaml .Values.global.securityContext | nindent 12 }}
+          {{- end }}
+          env:
```

Comment:
@pxsalehi please check if these env variable value are correct?

Reply:
If it's working with the normal installation path (not explicitly overriding the controller name), it should be good. You could try it also works when installed with a different name.